src/System.Runtime/tests/System/GCTests.netcoreapp.cs

```diff
@@ -35,5 +36,59 @@ public static void GetGCMemoryInfo()
             Assert.True(memoryInfo.HeapSizeBytes > 0);
             Assert.True(memoryInfo.FragmentedBytes >= 0);
         }
+
+        [Fact]
+        public static void GetGCMemoryInfoFragmentation()
+        {
+            GCHandle[] gch = null;
+
+            try
+            {
+                gch = new GCHandle[100];
+
+                for (int i = 0; i < gch.Length * 2; ++i)
+                {
+                    byte[] arr = new byte[128];
+                    if (i % 2 == 0)
+                    {
+                        gch[i / 2] = GCHandle.Alloc(arr, GCHandleType.Pinned);
+                    }
+                }
+
+                GC.Collect();
+
+                GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+                Assert.True(memoryInfo.FragmentedBytes >= 128 * (gch.Length - 1), $"FragmentedBytes = {memoryInfo.FragmentedBytes}");
+            }
+            finally
+            {
+                for (int i = 0; i < gch.Length; ++i)
+                {
+                    if (gch[i].IsAllocated)
+                    {
+                        gch[i].Free();
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public static void GetGCMemoryInfoHeapSize()
+        {
+            GCMemoryInfo memoryInfo1 = GC.GetGCMemoryInfo();
+
+            byte[][] arr = new byte[64 * 1024][];
+            for (int i = 0; i < arr.Length; ++i)
+            {
+                arr[i] = new byte[i + 1];
+            }
+
+            GCMemoryInfo memoryInfo2 = GC.GetGCMemoryInfo();
+
+            GC.KeepAlive(arr);
+
+            Assert.True(memoryInfo2.HeapSizeBytes > memoryInfo1.HeapSizeBytes);
```

Comment:
I think it'd be worth running this with:
```C#
RemoteInvoke(() =>
{
    ... // body of test you currently have
}).Dispose();
```
The System.Runtime.Tests, as with most test suites in corefx, run with parallelism enabled, so this test is likely to be running concurrently with a bunch of other System.Runtime tests, all of which could be allocating (along with xunit itself).  That means that it's theoretically possible for memoryInfo1 to represent a bunch of allocations performed in other tests that are then cleaned up by the time we snap memoryInfo2, in which case even with the allocations done in this test, in theory memoryInfo2 could be smaller than memoryInfo1.